### ISSUE TITLE
Delegate cannot be used now in the overridden container in v3.14.11 

### ISSUE THREAD

**Issue by GitterRemote** [ets-labs/python-dependency-injector#235]
```python
import collections                                                                 
from dependency_injector import containers                                         
from dependency_injector import providers                                          
                                                                                   
                                                                                   
A = collections.namedtuple('A', [])                                                
B = collections.namedtuple('B', ['fa'])                                            
C = collections.namedtuple('B', ['a'])  


class Services(containers.DeclarativeContainer):                                   
    a = providers.Dependency()                                                     
    c = providers.Factory(C, a=a)                                                  
    b = providers.Factory(B, fa=a.delegate())                                      
                                                                                   
                                                                                   
a = providers.Factory(A)                                                           
assert isinstance(Services(a=a).c().a, A)  # ok                                    
Services(a=a).b().fa()  # error: dependency_injector.errors.Error: Dependency is not defined
```

**Comment by rmk135** [comment]
Looking on it...

**Comment by rmk135** [comment]
Ok, I found it. `Delegate` provided was inherited from `Object` and got affected by `v3.14.11` changes.

Fix will come during the day.

**Comment by rmk135** [comment]
Fix is published in 3.14.12.

@GitterRemote , thanks a lot for reporting the issue.

Roman
